### DISCUSSION TITLE
Fixed Python Requirements

### DISCUSSION THREAD

**Discussion by blckbx** [mashb1t/Fooocus#19] [General]
Hey @mashb1t, 
was wondering if there is any obvious reason that I could be missing why we would want to fix python packages to distinct versions. Especially pytorch(-lightning) is kept on an older versions. Pytorch 2.2.0 reads like it's worth updating. 

Out of curiosity, I set Fooocus up without version pinning, instantly running into problems regarding gradio which were fixed by resetting gradio to recommended version 3.41.2. Using Pytorch 2.2.0 Fooocus is working well. Tested txt2img, img2img, upscaling, varying, in/outpainting, content modification. 

To be honest, it doesn't feel any faster. Nevertheless I wanted to point it out (specially pytorch).

**Comment by mashb1t** [comment]
Hey @blckbx,
mostly packages are not upgraded due to the many users in China or users in general with no possibility to access pypi. Packages will be upgraded when a significant feature is regarded as so impactful for them to force them to reinstall Fooocus completely from scratch (by downloading zhe zip file again) or when performance noticeably can be bumped.
Also see https://github.com/lllyasviel/Fooocus/discussions/2154#discussioncomment-8389770

**Comment by blckbx** [comment]
FWIW: I recently ran into problems after dist-upgrading Ubuntu to 24.04 LTS which comes with
```
Python 3.12.3
pip 24.0
```
I had to adjust package versions in `requirements_versions.txt` due to conflicts when building Fooocus like so:

```
torchsde
einops
transformers
safetensors
accelerate
pyyaml
Pillow
scipy
tqdm
psutil
pytorch_lightning
omegaconf
gradio==3.41.2
pygit2
opencv-contrib-python
diffusers
httpx
onnxruntime
timm
translators
rembg
groundingdino-py
```
